### PR TITLE
javascript-local-storage - Chivy Chia

### PR DESCRIPTION
**Pull request by tinachia**
image

### CODE REVIEW

File: javascript-local-storage/data.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* exported todos */
+
+var todos = [];
+var previousTodosJSON = localStorage.getItem('javascript-local-storage');
+todos = JSON.parse(previousTodosJSON);
```
Comment (Zhiwutian) [inline javascript-local-storage/data.js:5]: don't forget to do the check for `null` before assigning the parsed value to todos

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
small update needed

**Comment by tinachia** [comment]
image

File: javascript-local-storage/data.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+var todos = [];
+var previousTodosJSON = localStorage.getItem('javascript-local-storage');
+
+window.addEventListener('beforeunload', function (event) {
+  var todosJSON = JSON.stringify(todos);
+  localStorage.setItem('javascript-local-storage', todosJSON);
+  if (previousTodosJSON !== null) {
+    todos = JSON.parse(previousTodosJSON);
+  }
```
Comment (Zhiwutian) [inline javascript-local-storage/data.js:11]: this should be outside of the `beforeunload` function

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
small update needed!

**Comment by tinachia** [comment]
image

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**